pipeline/matcher/src/test/scala/uk/ac/wellcome/platform/matcher/matcher/WorkMatcherTest.scala

```diff
@@ -283,37 +261,30 @@ class WorkMatcherTest
     }
   }
 
-  it("throws MatcherException if it fails to unlock") {
-    withMockMetricsSender { mockMetricsSender =>
-      withWorkGraphTable { graphTable =>
-        withWorkGraphStore(graphTable) { workGraphStore =>
-          val mockDynamoRowLockDao = mock[DynamoRowLockDao]
-          withLockingService(mockDynamoRowLockDao, mockMetricsSender) {
-            dynamoLockingService =>
-              withWorkMatcherAndLockingService(
-                workGraphStore,
-                dynamoLockingService) { workMatcher =>
-                when(mockDynamoRowLockDao.lockRow(any[Identifier], any[String]))
-                  .thenReturn(
-                    Future.successful(
-                      RowLock(
-                        id = "id",
-                        contextId = "contextId",
-                        created = Instant.now,
-                        expires = Instant.now.plusSeconds(100))
-                    ))
-
-                when(mockDynamoRowLockDao.unlockRows(any[String])).thenReturn(
-                  Future.failed(
-                    FailedUnlockException("test", new RuntimeException)))
-
-                whenReady(
-                  workMatcher
-                    .matchWork(createUnidentifiedSierraWork)
-                    .failed) { failedMatch =>
-                  failedMatch shouldBe a[MatcherException]
-                }
-              }
+  // The new LockingService never passes on the information that an unlock has
+  // been unsuccessful (whether as an exception or returned from a method).
```

Comment:
[There is this](https://github.com/wellcometrust/scala-storage/blob/9bbfe64aec6871dae8f8113fe3329c7a561346d1/storage/src/main/scala/uk/ac/wellcome/storage/locking/LockingService.scala#L99).

Is it that the implementation isn't throwing this?

Reply:
That is never actually used:

```
> rg FailedUnlock
storage/src/main/scala/uk/ac/wellcome/storage/locking/LockingService.scala
99:case class FailedUnlock[ContextId, Ident](contextId: ContextId,
```
The [`unlock`](https://github.com/wellcometrust/scala-storage/blob/master/storage/src/main/scala/uk/ac/wellcome/storage/locking/LockingService.scala#L85) method returns a `Future[Unit]`, so it could be a failed future, but if you look at where `unlock` is called within that file it never does  anything with  the return value.